src/scripts/main.js

```diff
@@ -1,5 +1,33 @@
+/* eslint-disable no-shadow */
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+document.addEventListener('click', (event) => {
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  // Получить размеры и позиции стены и паука
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+
+  // Рассчитать центр паука
+  const spiderWidth = spiderRect.width;
+  const spiderHeight = spiderRect.height;
+
+  // Рассчитать новые координаты центра паука относительно клика
+  let newLeft = event.clientX - spiderWidth / 2;
+  let newTop = event.clientY - spiderHeight / 2;
```

Comment:
The current calculation for `newLeft` and `newTop` does not account for the page's scroll offset. You should use `event.pageX` and `event.pageY` instead of `event.clientX` and `event.clientY` to include the scroll offset in your calculations.